src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationEventSource.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics.Tracing;
+
+namespace System.Xml.Serialization
+{
+    [EventSource(Name = "System.Xml.Serialzation.XmlSerializationEventSource")]
+    internal class XmlSerializationEventSource : EventSource
+    {
+        internal static XmlSerializationEventSource Log = new XmlSerializationEventSource();
+
+        private const int TraceEventId = 1;
+
+        [Event(TraceEventId, Level = EventLevel.Informational)]
+        internal void Trace(string message)
```

Comment:
Look at [http://source.dot.net/#System.Private.ServiceModel/Internals/WcfEventSource.cs](http://source.dot.net/#System.Private.ServiceModel/Internals/WcfEventSource.cs) for a model of how to set this up. Create an inner class called EventIds and create a const int XmlSerializerExpired with the value 1. Change the method name to XmlSerializerExpired and give it two parameters: serializerName and type. In the attribute, set `Message = SR.XmlSerializerExpiredDetails`. Then in the body call WriteEvent(EventIds.XmlSerializerExpired, serializerName, type.FullName);